### PR TITLE
Fix code style

### LINKED ISSUE #117: Cleanup code to meet Dojo 2 style guide requirements
**Enhancement** 

Partially due to flux in the styleguide(requirement to avoid arrow functions changing to requirement to use arrow functions), and partially due to my inability to properly read, `stores` needs to be updated to match the current requirements of the [Dojo style guide](https://github.com/dojo/meta/blob/master/STYLE.md)

### PR DESCRIPTION
**Pull request by maier49**
The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

This updates the code to match the style guide. This partially addresses the concerns from #110 by adding `_` to protected and private variables where it was missing, but these changes only affect naming and formatting, not functionality. As such I didn't check whether any properties/methods should have their visibility changed, just made sure that they were ordered and named appropriately for the visibility they currently have.

Resolves #117

### CODE REVIEW

File: src/storage/InMemoryStorage.ts [hunk 1]
```diff
@@ -146,28 +103,59 @@ export default class InMemoryStorage<T> implements Storage<T, CrudOptions> {
...
+
+	identify(items: T[]| T): string[] {
+		const itemArray = Array.isArray(items) ? items : [ items ];
+		const idProperty = this._idProperty;
+		if (idProperty) {
+			return itemArray.map((item) => item[idProperty].toString());
+		}
+		else if (this._idFunction) {
+			return itemArray.map(this._idFunction);
+		}
```
Comment (agubler) [inline src/storage/InMemoryStorage.ts:136]: nit: don't need the `else` branch

File: src/storage/IndexedDBStorage.ts [hunk 2]
```diff
@@ -343,32 +348,32 @@ export default class IndexedDBStorage<T> extends InMemoryStorage<T> {
...
 			rejectInitialPromise = reject;
 		});
-		this.dbName = options.dbName || 'store-database';
-		this.objectStoreName = options.objectStoreName || 'items';
-		this.dbInitializationPromise = initializationPromise;
-		this.indices = new Set<keyof T>();
+		this._dbName = options.dbName || 'store-database';
+		this._objectStoreName = options.objectStoreName || 'items';
+		this._dbInitializationPromise = initializationPromise;
+		this._indices = new Set<keyof T>();
```
Comment (agubler) [inline src/storage/IndexedDBStorage.ts]: nit: properties like `this._indices` can be initialised in place - `private _indices = new Set<string>`.

File: src/store/ObservableStore.ts [hunk 3]
```diff
@@ -187,21 +191,21 @@ export interface ObservableStoreInterface<T, O extends CrudOptions, U extends Up
 	/**
 	 * Observe the entire store, receiving deltas indicating the changes to the store.
 	 * When observing, an initial update will be sent with the last known state of the store in the `afterAll` property.
-	 * If fetchAroundUpdates is true, the store's local data will by synchronized with the underlying Storage.
-	 * If fetchAroundUpdates is not true, then the data will be the result of locally applying updates to the data
+	 * If _fetchAroundUpdates is true, the store's local data will by synchronized with the underlying Storage.
```
Comment (agubler) [inline src/store/ObservableStore.ts]: Do we think we should be referencing internal private properties in API doc?

File: src/store/QueryResult.ts [hunk 4]
```diff
@@ -212,86 +216,100 @@ function transformItemUpdate<T, F>(
...
+
+	/**
+	 * Optional value that indicates the amount of time to debounce the fetch called after receiving an update.
+	 */
+	protected _trackingFetchDelay?: number;
+
+	/**
+	 * The store this query transform result comes from
+	 */
+	readonly source: S;
```
Comment (agubler) [inline src/store/QueryResult.ts]: should we be explicit about `public` visibility?

File: src/store/QueryResult.ts [hunk 5]
```diff
@@ -309,149 +327,184 @@ export class QueryResult<T, S extends QueryableStoreInterface<any, any, any>> im
...
+	 * @param ignoreNonIncrementalQueries
+	 * @returns {any[]}
+	 */
+	protected _queryAndTransformData(
+		data: T[],
+		queriesAndTransformations?: Array<Query<T> | TransformationDescriptor<T, any>>,
+		instance?: MappedQueryResultInterface<T, any>,
+		ignoreSorts = false,
+		ignoreNonIncrementalQueries = false
+	) {
```
Comment (agubler) [inline src/store/QueryResult.ts]: should we have the returns type here?

File: src/store/QueryResult.ts [hunk 6]
```diff
@@ -309,149 +327,184 @@ export class QueryResult<T, S extends QueryableStoreInterface<any, any, any>> im
...
-			return new MappedQueryResult<V, S>(options);
-		}
-		else {
-			return new QueryResult<V, S>(options);
+	/**
+	 * Sends the update if it actually represents any change in the data, and then removes observers that unsubscribed
+	 * from the list.
+	 * @param update
+	 */
+	protected _sendUpdate(update: StoreDelta<T>) {
```
Comment (agubler) [inline src/store/QueryResult.ts]: `: void`?

File: src/store/QueryResult.ts [hunk 7]
```diff
@@ -507,238 +560,204 @@ export class QueryResult<T, S extends QueryableStoreInterface<any, any, any>> im
...
+			else {
+				return this.source.get(ids)
+					.then((data) => {
+						if (Array.isArray(data)) {
+							return this._queryAndTransformData(data);
+						}
+						else if (data) {
+							return this._queryAndTransformData([data])[0];
+						}
+						else {
```
Comment (agubler) [inline src/store/QueryResult.ts]: can drop the `else`

File: src/store/QueryResult.ts [hunk 8]
```diff
@@ -507,238 +560,204 @@ export class QueryResult<T, S extends QueryableStoreInterface<any, any, any>> im
...
+				this._sourceHandle = waitForFetchPromise.then(() =>
+					this.source.observe()
+						.subscribe((update: StoreDelta<any>) => {
+							this._handleUpdate(update);
+						})
+				);
+			}
+			return this._observable;
+		}
+		else {
```
Comment (agubler) [inline src/store/QueryResult.ts]: can this be?

```ts
else if (Array.isArray(idOrIds)) {
    ...
}
return this.source
    .observe(idOrIds)
    .map((update: any) => transformData(this._queriesAndTransformations, update));
```

File: src/store/QueryResult.ts [hunk 9]
```diff
@@ -507,238 +560,204 @@ export class QueryResult<T, S extends QueryableStoreInterface<any, any, any>> im
...
+				...this._queriesAndTransformations,
+				{ transformation: transformation, idTransform: idTransform }
+			],
+			trackingFetchDelay: this._trackingFetchDelay,
+			fetchAroundUpdates: this._fetchAroundUpdates
 		};
+		if (idTransform) {
+			return new MappedQueryResult<V, S>(options);
+		}
+		else {
```
Comment (agubler) [inline src/store/QueryResult.ts]: nit: can drop `else`

File: tests/unit/storage/InMemoryStorage.ts [hunk 10]
```diff
@@ -93,60 +94,74 @@ registerSuite({
 		'Should put new items into storage.'(this: any) {
 			const { dfd, storage, data } = getStorageAndDfd(this);
 
-			storage.put(data).then(function(result) {
-				assert.deepEqual(result.successfulData, createData());
-			}).then(dfd.resolve);
+			storage.put(data)
```
Comment (agubler) [inline tests/unit/storage/InMemoryStorage.ts]: can't you just return the promise?

**Review by agubler — COMMENTED** [review]
Apologies if I am commenting on things that have not changed as part of this PR - It's difficult to really tell with all the green and reds :)

I have noted a few things, that generally could be addressed throughout the code - I'm happy if you want to address these after this code style PR has landed.

**Comment by maier49** [comment]
@agubler I've addressed all of the feedback. No objections to any of it.

File: src/storage/IndexedDBStorage.ts [hunk 11]
```diff
@@ -449,38 +484,29 @@ export default class IndexedDBStorage<T> extends InMemoryStorage<T> {
...
+				return results;
+			});
+	}
+
+	delete(ids: string[]): Promise<UpdateResults<T>> {
+		return this._dbInitializationPromise.then(() => {
+			const { transactionPromise, objectStore } = this._getTransactionAndObjectStore(true);
+			return Promise.all(ids.map((id) => createRequestPromise(
+				objectStore.delete(id), transactionPromise
+			)
```
Comment (agubler) [inline src/storage/IndexedDBStorage.ts]: this formatting seems a bit strange

**Review by agubler — APPROVED** [review]
Just a small formatting nit, but otherwise :+1:

File: src/storage/IndexedDBStorage.ts (hunk 11 above)
Comment (maier49) [reply src/storage/IndexedDBStorage.ts]: Doesn't count as nitpicking when the PR is to fix the formatting.

**Pull request merged**